hera_pspec/pspecdata.py

```diff
@@ -774,6 +779,25 @@ def R(self, key):
             elif self.data_weighting == 'iC':
                 self._R[Rkey] = sqrtT.T * sqrtY.T * self.iC(key) * sqrtY * sqrtT
 
+            elif self.data_weighting == 'sinc_downweight':
+                r_param_key = (self.data_weighting,) + key
+                if not r_param_key in self.r_params:
+                    raise_warning("Warning: no filter params specified for "
+                                    "sinc weights! Defaulting to Identity!")
```

Comment:
this is a situation where in the `history` of the output `uvp` object, the data weighting will be `'sinc_downweight'` but the power spectra will actually have `'identity'` weighting. I think we may be better off having this raise an Error rather than a warning. Perhaps @philbull has some thoughts on this.

Also, in theory we would want all of `self.r_params` to be propagated to the file history (b/c each baseline can have a different `r_param`) but the number of baselines is such that this will likely become too large. We should think of how we can pass through a minimum amount of information on these parameters through to the file history...

Reply:
As a minimum, you could include a hash of `self.r_params` in the history, as this would at least allow you to check whether the same settings are being used or not. It doesn't help you figure out what's different though...
And yes, it would be safer to raise an error instead of using Identity as default. If the user has mis-specified something, and it's not obvious what they meant to do, then we should stop them from proceeding :)

Reply:
Yes that is the point, but I think that instead of a warning and defaulting to `identity` we should raise an error: otherwise the output `uvp` objects will have `data_weighting='sinc_downweight'` in their metadata but will really have been weighted with `identity`. 

And also @anguta one way to propagate the `r_params` is to just propagate each unique `r_param` combination with its baseline-pair key: in a sense only propagating `r_params` for each unique redundant group, which would cut down on the size of the string you would be appending to the history. something to consider.

Reply:
I know throw an error when no `r_params` are specified.